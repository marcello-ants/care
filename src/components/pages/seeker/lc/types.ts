import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
