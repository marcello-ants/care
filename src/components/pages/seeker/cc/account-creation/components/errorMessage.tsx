import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
