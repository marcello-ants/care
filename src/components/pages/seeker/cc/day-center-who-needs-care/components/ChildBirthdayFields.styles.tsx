// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
