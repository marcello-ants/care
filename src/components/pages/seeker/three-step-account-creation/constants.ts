import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
