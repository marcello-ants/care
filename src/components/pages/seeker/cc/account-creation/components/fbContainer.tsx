import { Box, makeStyles, Grid, Button } from '@material-ui/core';
import { useFacebookApiKey } from '@/components/FacebookApiKeyContext';

// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { Icon48SocialMediaFacebook } from '@care/react-icons';

export type facebookResponseProps = {
  accessToken: string;
  // eslint-disable-next-line
  data_access_expiration_time?: number;
  email: string;
  expiresIn?: number;
  graphDomain?: string;
  id?: string;
  name: string;
  signedRequest?: string;
  userID?: string;
};

export interface SplitFormValues {
  email: string;
}

export interface SplitFormWithFbData extends SplitFormValues {
  firstName?: string;
  lastName?: string;
  accessToken?: string;
}

const mobileWidth = 360;
const fbColor = '#1877F2';
const fbIconSize = '35px';
const useStyles = (fbEmphasizedDesign: boolean) =>
  makeStyles((theme) => ({
    fbButton: {
      paddingTop: theme.spacing(5),
      borderColor: fbEmphasizedDesign ? fbColor : theme.palette.default?.dark,
      background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      color: fbEmphasizedDesign ? theme.palette.care?.white : theme.palette.default?.dark,
      fontSize: fbEmphasizedDesign ? '21px' : '18px',
      '&:hover': {
        background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      },
    },
    fbButtonContent: {
      display: 'grid',
      gridTemplateColumns: 'repeat(3, minmax(0, 1fr))',
      [theme.breakpoints.down(mobileWidth)]: {
        display: 'flex',
        alignItems: 'center',
      },
    },
    fbIconContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: fbIconSize,
      '& svg': {
        width: fbIconSize,
        height: fbIconSize,
      },
    },
    fbIcon: {
      background: '#1a77f2',
      borderRadius: '19px',
    },
    fbButtonText: {
      justifySelf: 'center',
      marginLeft: '10px',
      marginRight: '5px',
      fontColor: theme.palette.care?.white,
    },
  }));

type GetFbContainerProps = {
  handleFacebookLogin: (response: facebookResponseProps) => Promise<void>;
  fbButtonOnTop: boolean;
  fbButtonOnBottom: boolean;
};

const getFbContainer = ({
  handleFacebookLogin,
  fbButtonOnTop,
  fbButtonOnBottom,
}: GetFbContainerProps) => {
  const fbEmphasizedDesign = fbButtonOnBottom || fbButtonOnTop;
  const classes = useStyles(fbEmphasizedDesign)();
  const { apiKey } = useFacebookApiKey();

  return (
    <Grid item xs={12}>
      <Box pt={fbButtonOnTop ? 0 : 2}>
        <FacebookLogin
          appId={apiKey}
          autoLoad={false}
          fields="name,email"
          callback={handleFacebookLogin}
          render={(renderProps: any) => (
            <Button
              onClick={renderProps.onClick}
              className={classes.fbButton}
              variant="outlined"
              size="large"
              fullWidth>
              <div className={classes.fbButtonContent}>
                <div className={classes.fbIconContainer}>
                  <Icon48SocialMediaFacebook size="38px" color="white" className={classes.fbIcon} />
                </div>
                <p className={classes.fbButtonText}>Continue with Facebook</p>
              </div>
            </Button>
          )}
        />
      </Box>
    </Grid>
  );
};

export default getFbContainer;
