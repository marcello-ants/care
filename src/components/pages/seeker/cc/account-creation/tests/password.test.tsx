import React from 'react';
import { NextRouter } from 'next/router';
import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import userEvent from '@testing-library/user-event';
import { RouterContext } from 'next/dist/shared/lib/router-context';

import AuthService from '@/lib/AuthService';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import { BOOKING_MFE_IB_ASSESSMENT, CLIENT_FEATURE_FLAGS, FLOWS } from '@/constants';
import { CHANGE_MEMBER_PASSWORD } from '@/components/request/GQL';
import { TealiumUtagService } from '@/utilities/utagHelper';
import { DaycareProviderProfile, DefaultCareKind } from '@/types/seekerCC';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';

import DaycareAccountPassword from '@/components/pages/seeker/cc/account-creation/password';
import { cloneDeep } from 'lodash-es';

let mockRouter: NextRouter;
let buttonSetPassword: HTMLElement;
let buttonSkip: HTMLElement;
let inputPassword: HTMLElement;

const analyticsMock = jest
  .spyOn(AnalyticsHelper, 'logEvent')
  .mockImplementation(() => Promise.resolve());

const analyticsLogTestExposureMock = jest
  .spyOn(AnalyticsHelper, 'logTestExposure')
  .mockImplementation(() => Promise.resolve());

const TEST_FIRSTNAME = 'myfirst';
const TEST_LASTNAME = 'mylast';
const TEST_EMAIL = 'hello@care.com';
const TEST_PHONENUMBER = '(222) 2131415';
const VALID_PASSWORD = 'jayenclaospdh';
const INVALID_PASSWORD = 'password';
const GENERIC_PASSWORD_ERROR = 'generic error';

const skipLoginForRateCardVariation = {
  reason: { kind: 'kind' },
  value: true,
  variationIndex: 2,
};

jest.mock('next/config', () => () => {
  return {
    publicRuntimeConfig: {
      OIDC_STS_AUTHORITY: 'OIDC_STS_AUTHORITY',
      OIDC_CLIENT_ID: 'OIDC_CLIENT_ID',
      OIDC_RESPONSE_TYPE: 'OIDC_RESPONSE_TYPE',
      OIDC_CLIENT_SCOPE: 'OIDC_CLIENT_SCOPE',
      OIDC_CALLBACK_PATH: 'OIDC_CALLBACK_PATH',
      OIDC_LOGOUT_URL: 'OIDC_LOGOUT_URL',
      CZEN_GENERAL: 'https://www.dev.carezen.net',
    },
  };
});

jest.mock('@/utilities/utagHelper', () => ({
  __esModule: true,
  TealiumUtagService: {
    view: jest.fn(),
  },
}));
jest.mock('@/utilities/account-creation-utils', () => ({
  __esModule: true,
  hash256: jest.fn(() => Promise.resolve('testHash256')),
}));
const tealiumMock = TealiumUtagService.view as jest.Mock;

jest.mock('@/lib/AuthService');
const AuthServiceMock = AuthService as jest.Mock;
const windowLocationAssignMock = jest.fn();

const getStoreSpy = jest.fn();
const redirectLoginSpy = jest.fn();
AuthServiceMock.mockImplementation(() => {
  return {
    getStore: getStoreSpy,
    redirectLogin: redirectLoginSpy,
  };
});

function generateRecommendations(selectedValues: boolean[]): DaycareProviderProfile[] {
  const recommendations = [] as DaycareProviderProfile[];
  selectedValues.forEach((selected, idx) =>
    recommendations.push({
      __typename: 'Provider',
      id: `${idx}`,
      selected,
      name: '',
      avgReviewRating: 2,
      description: '',
      address: null,
      images: null,
      logo: null,
      license: null,
      reviews: null,
      hasCoordinates: true,
      centerType: null,
    })
  );
  return recommendations;
}

const changeMemberPasswordMockSuccess = {
  request: {
    query: CHANGE_MEMBER_PASSWORD,
    variables: {
      passwordInput: {
        password: VALID_PASSWORD,
      },
    },
  },
  result: {
    data: {
      memberChangePassword: {
        __typename: 'MemberChangePasswordSuccess',
        authToken: 'pPMAzhsSOJx5ZxZAy_e27*n28NNwBug4D0III1hM1Dw.',
      },
    },
  },
};

const changeMemberPasswordMockError = {
  request: {
    query: CHANGE_MEMBER_PASSWORD,
    variables: {
      passwordInput: {
        password: INVALID_PASSWORD,
      },
    },
  },
  result: {
    data: {
      memberChangePassword: {
        __typename: 'MemberChangePasswordError',
        errors: [{ __typename: 'MemberPasswordInvalid', message: GENERIC_PASSWORD_ERROR }],
      },
    },
  },
};

const defaultCarekind = {
  careKind: DefaultCareKind.DAY_CARE_CENTERS,
};

function renderPage(mocks: MockedResponse[] = [], appState?: AppState, ldFlags?: FeatureFlags) {
  getStoreSpy.mockReturnValue({ profile: { email: TEST_EMAIL } });
  const initialState: AppState = appState || {
    ...initialAppState,
    seekerCC: {
      ...initialAppState.seekerCC,
      ...defaultCarekind,
      firstName: TEST_FIRSTNAME,
      lastName: TEST_LASTNAME,
      phoneNumber: TEST_PHONENUMBER,
      isSendLeadEnabled: true,
    },
    flow: {
      ...initialAppState.flow,
      flowName: FLOWS.SEEKER_CHILD_CARE.name,
    },
  };

  // @ts-ignore
  mockRouter = {
    push: jest.fn(),
    asPath: '',
    pathname: '',
    basePath: '',
  };
  const view = render(
    <RouterContext.Provider value={mockRouter}>
      <MockedProvider mocks={mocks} addTypename={false}>
        <FeatureFlagsProvider flags={ldFlags || {}}>
          <AppStateProvider initialStateOverride={initialState}>
            <DaycareAccountPassword />
          </AppStateProvider>
        </FeatureFlagsProvider>
      </MockedProvider>
    </RouterContext.Provider>
  );

  buttonSetPassword = screen.getByRole('button', { name: 'Set password' });
  buttonSkip = screen.getByText('Skip for now');
  inputPassword = screen.getByLabelText('Password');

  return view;
}

describe('Day Care Account Password - Update Password', () => {
  const originalLocation = window.location;

  beforeAll(() => {
    // @ts-ignore
    delete window.location;
    // @ts-ignore
    window.location = {
      assign: windowLocationAssignMock,
    };
  });

  afterAll(() => {
    window.location = originalLocation;
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderPage();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders password field, set password and skip buttons', () => {
    renderPage();
    expect(inputPassword).toBeInTheDocument();
    expect(buttonSetPassword).toBeInTheDocument();
    expect(buttonSkip).toBeInTheDocument();
  });

  it('renders the account created banner', () => {
    renderPage();
    expect(screen.getByText('Account created!')).toBeInTheDocument();
  });

  it('pressing "skip for now" redirects user to next page', () => {
    renderPage();
    userEvent.click(buttonSkip);
    expect(windowLocationAssignMock).toHaveBeenCalledWith(
      'https://www.dev.carezen.net/vis/auth/login?forwardUrl=https%3A%2F%2Fwww.dev.carezen.net%2Fapp%2Fratecard%2Fchildcare%2Fpre-rate-card%3FenrollFlow%3Dtrue'
    );
  });

  it('fires test exposure event - A/B test - skip-login-in-ratecard-redirection', () => {
    renderPage(undefined, undefined, {
      [CLIENT_FEATURE_FLAGS.SKIP_LOGIN_IN_RATECARD_REDIRECTION]: skipLoginForRateCardVariation,
    });

    expect(analyticsLogTestExposureMock).toBeCalledWith(
      'skip-login-in-ratecard-redirection',
      skipLoginForRateCardVariation
    );
  });

  it('pressing "skip for now" redirects user directly to ratecard-mfe - A/B test - skip-login-in-ratecard-redirection', () => {
    renderPage(undefined, undefined, {
      [CLIENT_FEATURE_FLAGS.SKIP_LOGIN_IN_RATECARD_REDIRECTION]: skipLoginForRateCardVariation,
    });
    userEvent.click(buttonSkip);

    expect(windowLocationAssignMock).toHaveBeenCalledWith(
      'https://www.dev.carezen.net/app/ratecard/childcare/pre-rate-card?enrollFlow=true'
    );
  });

  it('pressing "set password" redirects user directly to ratecard-mfe - A/B test - skip-login-in-ratecard-redirection', async () => {
    renderPage([changeMemberPasswordMockSuccess], undefined, {
      [CLIENT_FEATURE_FLAGS.SKIP_LOGIN_IN_RATECARD_REDIRECTION]: skipLoginForRateCardVariation,
    });

    userEvent.type(inputPassword, VALID_PASSWORD);
    inputPassword.blur();
    userEvent.click(buttonSetPassword);

    await waitFor(() => {
      expect(windowLocationAssignMock).toHaveBeenCalledWith(
        'https://www.dev.carezen.net/app/ratecard/childcare/pre-rate-card?enrollFlow=true'
      );
    });
  });

  it('validates password is at least 6 characters long', async () => {
    renderPage();

    userEvent.type(inputPassword, '123');
    inputPassword.blur();

    expect(await screen.findByText(/Password must be at least 6 characters./i)).toBeInTheDocument();
  });

  it('validates password does not contain first name', async () => {
    renderPage();

    userEvent.type(inputPassword, TEST_FIRSTNAME);
    inputPassword.blur();

    expect(
      await screen.findByText(/Please don't use your first name in your password./i)
    ).toBeInTheDocument();
  });

  it('validates password does not contain last name', async () => {
    renderPage();

    userEvent.type(inputPassword, TEST_LASTNAME);
    inputPassword.blur();

    expect(
      await screen.findByText(/Please don't use your last name in your password./i)
    ).toBeInTheDocument();
  });

  it('validates password does not match email', async () => {
    renderPage();

    userEvent.type(inputPassword, TEST_EMAIL);
    inputPassword.blur();

    expect(
      await screen.findByText(/Please don't use your name or email address in your password./i)
    ).toBeInTheDocument();
  });

  it(`pressing "set password" updates user's password and redirects user to next page`, async () => {
    renderPage([changeMemberPasswordMockSuccess]);

    userEvent.type(inputPassword, VALID_PASSWORD);
    inputPassword.blur();
    userEvent.click(buttonSetPassword);

    await waitFor(() => {
      expect(windowLocationAssignMock).toHaveBeenCalledWith(
        'https://www.dev.carezen.net/vis/auth/login?forwardUrl=https%3A%2F%2Fwww.dev.carezen.net%2Fapp%2Fratecard%2Fchildcare%2Fpre-rate-card%3FenrollFlow%3Dtrue'
      );
    });
  });

  it('renders error banner when there is an error', async () => {
    renderPage([changeMemberPasswordMockError]);
    userEvent.type(inputPassword, INVALID_PASSWORD);
    inputPassword.blur();
    userEvent.click(buttonSetPassword);

    expect(await screen.findByText(GENERIC_PASSWORD_ERROR)).toBeInTheDocument();
  });

  it('sends the right value for `leadCount` in the Tealium event', () => {
    const slots = { slots: ['/us-marketplace/daycare/leads-submitted/'] };
    const appState: AppState = {
      ...initialAppState,
      seekerCC: {
        ...initialAppState.seekerCC,
        ...defaultCarekind,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations: generateRecommendations([true, false, true, false]),
        },
        firstName: TEST_FIRSTNAME,
        lastName: TEST_LASTNAME,
        phoneNumber: TEST_PHONENUMBER,
      },
      flow: {
        ...initialAppState.flow,
        flowName: FLOWS.SEEKER_CHILD_CARE.name,
      },
    };
    renderPage([], appState);
    expect(tealiumMock).not.toHaveBeenCalledWith(slots);
  });

  it('useTealiumTracking should be called when careKind is DAYCARE', async () => {
    renderPage();
    await waitFor(() => {
      expect(tealiumMock).toHaveBeenCalledTimes(1);
    });
  });

  it('useTealiumTracking should not be called when careKind is ONE_TIME_BABYSITTERS', async () => {
    const slots = [
      '/us-subscription/conversion/seeker/basic/signup/',
      '/us-subscription/conversion/seeker/basic/signup/impact/',
    ];
    const teliumData = {
      sessionId: undefined,
      slots,
      tealium_event: 'CONGRATS_BASIC_MEMBERSHIP',
      email: TEST_EMAIL,
      emailSHA256: 'testHash256',
      intent: 'WITHIN_A_WEEK',
      memberType: 'seeker',
      overallStatus: 'basic',
      serviceId: 'CHILD_CARE',
      subServiceId: 'ONE_TIME_BABYSITTERS',
    };
    defaultCarekind.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
    renderPage();

    await waitFor(() => {
      expect(tealiumMock).toBeCalledWith(teliumData);
    });
  });

  it('careKind is not DAY_CARE_CENTERS pressing "skip for now" redirects user to second next page', () => {
    defaultCarekind.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
    renderPage();
    userEvent.click(buttonSkip);
    expect(windowLocationAssignMock).toHaveBeenCalledWith('/seeker/cc/schedule');
  });

  it(`careKind is not DAY_CARE_CENTERS, pressing "set password" updates user's password, redirects user to next page`, async () => {
    defaultCarekind.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
    renderPage([changeMemberPasswordMockSuccess]);
    userEvent.type(inputPassword, VALID_PASSWORD);
    inputPassword.blur();
    userEvent.click(buttonSetPassword);

    userEvent.type(inputPassword, VALID_PASSWORD);
    inputPassword.blur();
    userEvent.click(buttonSetPassword);

    await waitFor(() => {
      expect(windowLocationAssignMock).toHaveBeenCalledWith('/seeker/cc/schedule');
    });
  });

  it('leads-submitted tealium and amplitude should not be called when recommendations is empty', () => {
    const slots = { slots: ['/us-marketplace/daycare/leads-submitted/'] };
    const appState: AppState = {
      ...initialAppState,
      seekerCC: {
        ...initialAppState.seekerCC,
        ...defaultCarekind,
        dayCare: {
          ...initialAppState.seekerCC.dayCare,
          recommendations: generateRecommendations([]),
        },
        firstName: TEST_FIRSTNAME,
        lastName: TEST_LASTNAME,
        phoneNumber: TEST_PHONENUMBER,
      },
      flow: {
        ...initialAppState.flow,
        flowName: FLOWS.SEEKER_CHILD_CARE.name,
      },
    };
    renderPage([], appState);
    expect(tealiumMock).not.toHaveBeenCalledWith(slots);
    expect(analyticsMock).not.toHaveBeenCalledWith({
      name: 'Slot Sent',
      data: slots,
    });
  });

  it('Basic signup tealium and amplitude should be call on page load', async () => {
    const slots = [
      '/us-subscription/conversion/seeker/basic/signup/',
      '/us-subscription/conversion/seeker/basic/signup/impact/',
    ];
    const teliumData = {
      sessionId: undefined,
      slots,
      tealium_event: 'CONGRATS_BASIC_MEMBERSHIP',
      email: TEST_EMAIL,
      emailSHA256: 'testHash256',
      intent: 'WITHIN_A_WEEK',
      memberType: 'seeker',
      overallStatus: 'basic',
      serviceId: 'CHILD_CARE',
      subServiceId: 'ONE_TIME_BABYSITTERS',
    };
    renderPage();

    await waitFor(() => {
      expect(tealiumMock).toHaveBeenCalledWith(teliumData);
    });
    await waitFor(() => {
      expect(analyticsMock).toHaveBeenCalledWith({
        name: 'Slot Sent',
        data: {
          slots,
        },
      });
    });
  });
  it('should render with SEEKER_CC_CONVERSATIONAL_LANGUAGE text', () => {
    renderPage([], initialAppState, {
      [CLIENT_FEATURE_FLAGS.SEEKER_CC_CONVERSATIONAL_LANGUAGE]: {
        variationIndex: 2,
        value: 2,
        reason: { kind: '' },
      },
    });
    expect(screen.getByText(/Set a password for next time./)).toBeInTheDocument();
  });
  it('should render without SEEKER_CC_CONVERSATIONAL_LANGUAGE text', () => {
    renderPage([], initialAppState, {
      [CLIENT_FEATURE_FLAGS.SEEKER_CC_CONVERSATIONAL_LANGUAGE]: {
        variationIndex: 0,
        value: 0,
        reason: { kind: '' },
      },
    });
    expect(screen.getByText('Finish setting up your account')).toBeInTheDocument();
  });

  it('should render with PASSWORD_REQUIREMENTS_COPY text', () => {
    renderPage([], initialAppState, {
      [CLIENT_FEATURE_FLAGS.PASSWORD_REQUIREMENTS_COPY]: {
        variationIndex: 0,
        value: 2,
        reason: { kind: '' },
      },
    });
    expect(
      screen.getByText(
        'Create a new password to save time, or skip it for now, and we’ll email you a temporary one'
      )
    ).toBeInTheDocument();
  });

  it('should render without PASSWORD_REQUIREMENTS_COPY text', () => {
    renderPage([], initialAppState, {
      [CLIENT_FEATURE_FLAGS.PASSWORD_REQUIREMENTS_COPY]: {
        variationIndex: 0,
        value: 1,
        reason: { kind: '' },
      },
    });
    expect(
      screen.queryByText(
        'Create a new password to save time, or skip it for now, and we’ll email you a temporary one'
      )
    ).not.toBeInTheDocument();
  });

  describe('Instant book redirection flow', () => {
    let ldFlags: FeatureFlags;
    const redirectURL = `https://www.dev.carezen.net${BOOKING_MFE_IB_ASSESSMENT}`;

    beforeEach(() => {
      ldFlags = {
        [CLIENT_FEATURE_FLAGS.INSTANT_BOOK_ENROLLMENT_M1]: {
          variationIndex: 2,
          value: '2',
          reason: { kind: '' },
        },
      };
    });

    function skipRedirect(redirectsToInstantBookFlow: boolean) {
      userEvent.click(buttonSkip);
      if (redirectsToInstantBookFlow) {
        expect(windowLocationAssignMock).toHaveBeenCalledWith(redirectURL);
      } else {
        expect(windowLocationAssignMock).not.toHaveBeenCalledWith(redirectURL);
      }
    }

    async function setPasswordRedirect(redirectsToInstantBookFlow: boolean) {
      // set password
      userEvent.type(inputPassword, VALID_PASSWORD);
      inputPassword.blur();
      userEvent.click(buttonSetPassword);

      await waitFor(() => {
        if (redirectsToInstantBookFlow) {
          expect(windowLocationAssignMock).toHaveBeenCalledWith(redirectURL);
        } else {
          expect(windowLocationAssignMock).not.toHaveBeenCalledWith(redirectURL);
        }
      });
    }

    async function testRedirectURL(redirectsToInstantBookFlow = false) {
      // test skip
      skipRedirect(redirectsToInstantBookFlow);
      // test set password
      await setPasswordRedirect(redirectsToInstantBookFlow);
    }

    it('should not redirect to booking MFE when careKind is not ONE_TIME_BABYSITTERS (skip or set pass btn)', async () => {
      const appState: AppState = cloneDeep(initialAppState);
      appState.seekerCC.instantBook.eligibleLocation = true;
      appState.seekerCC.careKind = DefaultCareKind.DAY_CARE_CENTERS;
      renderPage([changeMemberPasswordMockSuccess], appState, ldFlags);

      await testRedirectURL();
    });
    it('should not redirect to booking MFE when location is not eligible (skip or set pass btn)', async () => {
      const appState: AppState = cloneDeep(initialAppState);
      appState.seekerCC.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
      appState.seekerCC.instantBook.eligibleLocation = false;
      renderPage([changeMemberPasswordMockSuccess], appState, ldFlags);

      await testRedirectURL();
    });
    it('should not redirect to booking MFE when instantBookM1Flag variation is not 2 (skip or set pass btn)', async () => {
      const appState: AppState = cloneDeep(initialAppState);
      appState.seekerCC.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
      appState.seekerCC.instantBook.eligibleLocation = true;

      ldFlags[CLIENT_FEATURE_FLAGS.INSTANT_BOOK_ENROLLMENT_M1]!.variationIndex = 0;
      renderPage([changeMemberPasswordMockSuccess], appState, ldFlags);

      await testRedirectURL();
    });
    it('should redirect to booking MFE when all criteria above passed (skip or set pass btn)', async () => {
      const appState: AppState = cloneDeep(initialAppState);
      appState.seekerCC.careKind = DefaultCareKind.ONE_TIME_BABYSITTERS;
      appState.seekerCC.instantBook.eligibleLocation = true;

      renderPage([changeMemberPasswordMockSuccess], appState, ldFlags);

      await testRedirectURL(true);
    });
  });
});
