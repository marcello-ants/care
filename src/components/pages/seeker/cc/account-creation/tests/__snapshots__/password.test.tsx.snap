// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Care Account Password - Update Password matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="makeStyles-root-4"
    >
      <div
        class="makeStyles-content-5 makeStyles-confirmation-12 makeStyles-roundCorners-7"
        style="width: 100%;"
      >
        <div
          class="makeStyles-icon-8"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-root-38"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0A8.68 8.68 0 1012 3.32 8.7 8.7 0 003.32 12z"
              fill="#00a979"
            />
            <path
              d="M10.35 15.3a.68.68 0 01-.36-.18l-2.43-2.64a.54.54 0 01.09-.62.45.45 0 01.63 0l2.07 2.29 4.78-5a.45.45 0 01.63 0 .41.41 0 010 .62l-5 5.37a.49.49 0 01-.41.16z"
              fill="#00a979"
            />
            <path
              d="M10.35 15.55a.91.91 0 01-.53-.25l-2.48-2.71a.77.77 0 01.13-.9.7.7 0 011 0l1.9 2.11L15 9a.7.7 0 011 0 .68.68 0 01.21.49.67.67 0 01-.21.49l-5 5.37a.77.77 0 01-.65.2zm-2.58-3.21L10.18 15a.54.54 0 00.18.1c.07 0 .1 0 .18-.1l5-5.37a.19.19 0 00.07-.14.18.18 0 00-.06-.13.19.19 0 00-.28 0l-5 5.21L8.1 12a.2.2 0 00-.28 0 .28.28 0 00-.05.34z"
              fill="#00a979"
            />
          </svg>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Account created!
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiBox-root MuiBox-root-29"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h2
          class="MuiTypography-root makeStyles-header-30 MuiTypography-h2"
        >
          Finish setting up your account
        </h2>
      </div>
      <div
        class="MuiBox-root MuiBox-root-31"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          We created a temporary password, but you can set a new one now to save time.
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-inputContainer-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inline-32"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-describedby="password-helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root"
          id="password-helper-text"
        >
          Minimum 6 characters
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-36"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Set password
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiBox-root MuiBox-root-37"
      >
        <h6
          class="MuiTypography-root makeStyles-skipForNowLink-1 MuiTypography-subtitle1"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
          >
            Skip for now
          </a>
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
